### PR TITLE
New helm chart and documentation 

### PR DESCRIPTION
**Pull request by name212**
Move helm_lib directory from deckhouse repo to standalone repo:
- Add documentation generator
- Add helm repo generator based on `helm/chart-releaser-action`

### CODE REVIEW

File: charts/helm_lib/README.md [hunk 1]
```diff
@@ -0,0 +1,847 @@
...
+
+## High Availability
+
+### helm_lib_is_ha_to_value
+
+ returns value <yes> if cluster is highly available, else — returns <no> 
+
+#### Usage
+
+`{{ include "helm_lib_is_ha_to_value" (list . <yes> <no>) }} `
```
Comment (shvgn) [inline charts/helm_lib/README.md:104]: Please, don't use angle brackets in Markdown or escape them. I believe angle brackets will hurt us again and again, so I'd choose other brackets. HTML is rendered in Markdown.

See, it disappears: 

> returns value <yes> if cluster is highly available, else — returns <no> 

vs

> returns value `<yes>` if cluster is highly available, else — returns `<no>`

Non-HTML brackets.

> returns value {yes} if cluster is highly available, else — returns {no}

Screenshot:

<hr />

image

File: charts/helm_lib/README.md [hunk 2]
```diff
@@ -0,0 +1,847 @@
...
+
+#### Arguments
+
+list:
+-  Dot object (.) with .Values, .Chart, etc
```
Comment (shvgn) [inline charts/helm_lib/README.md:109]: "Dot object" is just a "template context". I'd rephrase it everywhere

File: charts/helm_lib/README.md [hunk 3]
```diff
@@ -0,0 +1,847 @@
+# Helm utils template definitions for Deckhouse modules
```
Comment (shvgn) [inline charts/helm_lib/README.md:1]: "Helm utils template definitions" could be just "Helm library" because there cannot be anything more than reusable definitions AFAIK.

```suggestion
# Helm library for Deckhouse modules
```

File: charts/helm_lib/README.md [hunk 4]
```diff
@@ -0,0 +1,847 @@
...
+
+## Table of contents
+
+| Table of contents |
```
Comment (shvgn) [inline charts/helm_lib/README.md:5]: The ToC could be just a two-level nested list. I believe it is easier to read than one-level mixture of headings and links

```
- Envs For Proxy
  - [helm_lib_envs_for_proxy](#helm_lib_envs_for_proxy)
- High Availability
  - [helm_lib_is_ha_to_value](#helm_lib_is_ha_to_value)
  - [helm_lib_ha_enabled](#helm_lib_ha_enabled)
```

- Envs For Proxy
  - [helm_lib_envs_for_proxy](#helm_lib_envs_for_proxy)
- High Availability
  - [helm_lib_is_ha_to_value](#helm_lib_is_ha_to_value)
  - [helm_lib_ha_enabled](#helm_lib_ha_enabled)

**Review by shvgn — CHANGES_REQUESTED** [review]

**Comment by shvgn** [comment]
By the way we call out definitions "helm_lib_*" which is too generic. We could namespace the definitions more precisely if we call them with e.g. "d8_" prefix

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
[the remark this answers is not in the document]
Comment (name212) [reply LICENSE:1]: First file located in root of repository, I think root of repository should contain license file, but t am not sure.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,7 @@
+.PHONY : doc/build
+
+doc/build:
+	go run tools/build-doc.go
+
+doc/diff:
+	go run tools/build-doc.go --diff
```
[the remark this answers is not in the document]
Comment (name212) [reply Makefile:7]: I foget

File: charts/helm_lib/README.md (hunk 3 above)
Comment (name212) [reply charts/helm_lib/README.md]: done

File: charts/helm_lib/README.md (hunk 2 above)
Comment (name212) [reply charts/helm_lib/README.md]: fixed

File: charts/helm_lib/README.md (hunk 1 above)
Comment (name212) [reply charts/helm_lib/README.md]: fixed

File: charts/helm_lib/README.md (hunk 4 above)
Comment (name212) [reply charts/helm_lib/README.md:5]: We decided keep table for first iteration.

File: LICENSE (hunk 5 above)
[the remark this answers is not in the document]
Comment (kvaps) [reply LICENSE:1]: Good point, I like the idea of having LICENSE file in root directory.
But I'm not sure if we can place it there while repository contains other code under other license

**Review by shvgn — APPROVED** [review]
